solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformer.java

```diff
@@ -132,6 +134,12 @@ public void transform(SolrDocument rootDoc, int rootDocId) {
 
           // load the doc
           SolrDocument doc = searcher.getDocFetcher().solrDoc(docId, childReturnFields);
+          if(childReturnFields.getTransformer() != null) {
+            if(childReturnFields.getTransformer().context != this.context) {
```

Comment:
maybe this line should only check if context is null?  By checking if the context is different, it's suggestive there may already be a context but if that were true then we ought to reinstate the former context when done.  I suspect it's always going to be null.  Can you look?